contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol

```diff
@@ -714,6 +715,16 @@ abstract contract KeeperRegistryBase2_1 is ConfirmedOwner, ExecutionPrevention {
     }
   }
 
+  /**
+   * @notice returns a unique identifier for an upkeep/trigger combo
+   * @param upkeepID the upkeep id
+   * @param trigger the raw trigger bytes
+   * @return triggerID the unique identifier for the upkeep/trigger combo
+   */
+  function _triggerID(uint256 upkeepID, bytes memory trigger) internal pure returns (bytes32 triggerID) {
```

Comment:
not sure if this will be automatically exposed in gethwrapper? I think this will be needed offchain to compute the trigger ID to listen to in logs. cc @amirylm

Reply:
It's internal so won't be exposed, but offchain should be able to compute the triggerID without making an rpc call by just replicating the logic. I think that's probably cleanest to just duplicate the function in go offchain.

Reply:
ah i was thinking more in terms of exposing the abi.encoding through gethwrapper easily but I see there's a keccak256 also which will need to be replicated so nvm